Sources/TUSKit/TUSClient.swift

```diff
@@ -369,20 +373,53 @@ public final class TUSClient {
         self.scheduler.checkProcessNextTasks()
     }
     
+    /// Background tasks don't communicate progress back to react-native-tus
+    /// This method allows react-native app to sync with the metadata filesystem
+    @discardableResult
+    public func sync() -> [[String:Any]] {
+        do {
+            if(updatesToSync.count == 0) {
+                try getUpdatesToSync()
+            }
+            let updates = updatesToSync.map { update in
+                return [
+                  "id": "\(update.id)",
+                  "bytesUploaded": update.bytesUploaded,
+                  "size": update.size,
+                  "isError": update.errorCount >= retryCount,
+                  "name": update.name
+                ]
+            }
+            updatesToSync.removeAll()
+            return updates
+        } catch let error {
+            return []
+        }
+    }
+    
     // MARK: - Private
     
-    /// Check for any uploads that are finished and remove them from the cache.
-    private func removeFinishedUploads() {
+    /// Builds a list of files and their current status so parent can stay in sync with TUSClient
+    /// Also, checks for any uploads that are finished and remove them from the cache (Background uploads don't have clean up)
+    private func getUpdatesToSync() {
         do {
-            let metaDataList = try files.loadAllMetadata()
-                .filter { metaData in
-                    metaData.size == metaData.uploadedRange?.count
+            try files.loadAllMetadata()
+            .forEach{ metaData in
+                let tusClientUpdate = TUSClientUpdate(id: metaData.id,
+                                                      bytesUploaded: metaData.uploadedRange?.count ?? 0,
+                                                      size: metaData.size,
+                                                      errorCount: metaData.errorCount,
+                                                      name: metaData.context?["name"] ?? "")
+                if (metaData.size == metaData.uploadedRange?.count) {
+                    try files.removeFileAndMetadata(metaData)
                 }
-            
-            for metaData in metaDataList {
-                try files.removeFileAndMetadata(metaData)
+                updatesToSync.append(tusClientUpdate)
             }
         } catch let error {
+            /// If called and no background tasks exist this error occurs
+            if (error.localizedDescription == "The file “background” couldn’t be opened because there is no such file.") {
```

Comment:
This error message is specific to our implementation of RN Tus. We ask TUSKit to create a directory at TUS/background.

Reply:
I was getting this error message randomly while testing. Any idea why the directory wouldn't exist? I assume this needs to be removed as if the dir doesn't exist and it's supposed to, then bad things...